src/dft/backends/mklcpu/commit.cpp

```diff
@@ -28,22 +28,94 @@
 
 #include "oneapi/mkl/dft/detail/mklcpu/onemkl_dft_mklcpu.hpp"
 
+#include "oneapi/mkl/dft/detail/commit_impl.hpp"
+#include "mkl_service.h"
+#include "mkl_dfti.h"
+
 namespace oneapi {
 namespace mkl {
 namespace dft {
 namespace mklcpu {
 
-void commit_f(descriptor<precision::SINGLE, domain::REAL> &desc, sycl::queue &queue) {
-    throw std::runtime_error("Not implemented for mklcpu");
-}
-void commit_c(descriptor<precision::SINGLE, domain::COMPLEX> &desc, sycl::queue &queue) {
-    throw std::runtime_error("Not implemented for mklcpu");
-}
-void commit_d(descriptor<precision::DOUBLE, domain::REAL> &desc, sycl::queue &queue) {
-    throw std::runtime_error("Not implemented for mklcpu");
-}
-void commit_z(descriptor<precision::DOUBLE, domain::COMPLEX> &desc, sycl::queue &queue) {
-    throw std::runtime_error("Not implemented for mklcpu");
+class commit_derived_impl : public oneapi::mkl::dft::detail::commit_impl {
+public:
+    commit_derived_impl(sycl::queue queue, dft_values config_values)
+            : oneapi::mkl::dft::detail::commit_impl(queue),
+              status(false) {
+
+        DFTI_DESCRIPTOR_HANDLE local_handle = nullptr;
+
+        if (config_values.rank == 1) {
+            status = DftiCreateDescriptor(&local_handle, get_precision(config_values.precision),
+                                          get_domain(config_values.domain), config_values.rank,
+                                          config_values.dimensions[0]);
+        } else {
+            status = DftiCreateDescriptor(&local_handle, get_precision(config_values.precision),
+                                          get_domain(config_values.domain), config_values.rank,
+                                          config_values.dimensions.data());
+        }
+        if(status != DFTI_NO_ERROR) {
```

Comment:
Isn't `status` a bool? 

This works with `status != DFTI_NO_ERROR`, but won't work for specific errors. Perhaps we ought to make status part of the backend-specific derived class so it can be of a type appropriate to the errors created by that backend.

Reply:
Fair point, mostly the error checking would be done after each create, commit, compute and can be handled trivally, but for set_value
```
status |= DftiSetValue(descHandle, DFTI_INPUT_STRIDES, ...);
status |= DftiSetValue(descHandle, DFTI_OUTPUT_STRIDES, ...);
status |= DftiSetValue(descHandle, DFTI_BACKWARD_SCALE, ...);
```
the value of status would be polluted if there are different return types of the (cpu) SetValue. so we can handle this by:
1. expanding `dft_values` struct to store return values of each named constant.
2. introduce erorr checks after each `DftiSetValue`
3. introduce vector/map of error status and dump a list of all error if any_value is not `DFTI_NO_ERROR`
for now just changig the bool to a int

Reply:
I think ideally we should check between each `DftiSetValue` and throw an appropiate exception with a helpful message if an error is encountered. Perhaps wrap DtfiSetValue in a function that helps with this?